webpack/js/plugins.js

```diff
@@ -0,0 +1,34 @@
+const webpack = require('webpack');
+
+/* NOTE:
+ * Do not require UglifyJS, found a significant slowdown in build times
+ * but no difference between file sizes
+ */
+module.exports = (ENV) => {
+  const uglifyPlugin = (ENV === 'production')
+    ? new webpack.optimize.UglifyJsPlugin({
+      comments: false,
+    })
+    : null;
+
+  return [
+    new webpack.ProvidePlugin({
+      jQuery: 'jquery',
+      $: 'jquery',
+    }),
+    new webpack.DefinePlugin({
+      'process.env': {
+        // Builds React in production mode, avoiding console warnings
+        NODE_ENV: JSON.stringify(ENV),
+      },
+    }),
+    // Most vendor libs are loaded directly into the 'vendor' bundle (through require()
+    // calls in vendor.js). This ensures that any further require() calls in other
+    // bundles aren't duplicating libs.
+    new webpack.optimize.CommonsChunkPlugin({
+      name: 'vendor',
+      minChunks: module => module.context && module.context.indexOf('/node_modules/') > -1,
+    }),
+    uglifyPlugin,
+  ].filter(plugin => plugin);
```

Comment:
`uglifyPlugin` is null in non-production builds, uglifyPlugin is only passed in to remove all comments from the `dist` build file.